src/include/error.h

```diff
@@ -128,9 +128,34 @@
 		__wt_abort(session);					\
 	}								\
 } while (0)
+#define	WT_ERR_ASSERT(session, exp, v, ...) do {			\
+	if (!(exp)) {							\
+		ret = (v);						\
+		__wt_err(session, ret, __VA_ARGS__);			\
+		__wt_abort(session);					\
+	}								\
+} while (0)
+#define	WT_RET_ASSERT(session, exp, v, ...) do {			\
+	if (!(exp))							\
+	{								\
+		int __ret = (v);					\
+		__wt_err(session, __ret, __VA_ARGS__);			\
+		__wt_abort(session);					\
+	}								\
+} while (0)
```

Comment:
I don't think we need the `ret` or `__ret` temporary variables, and the names are a little odd (why `ret` in one case, and `__ret` in the other?)